### PR TITLE
premint v2 sdk

### PR DESCRIPTION
**Pull request by oveddan**
Premint sdk now supports premint v2. See the added changeset for a summary of changes.

ToDo:
* [ ] Check when creating premints, that the premint version is supported by the contract.  if v2 is supported by the contract, then require premint to be created as a v2 signature

### CODE REVIEW

**Comment by oveddan** [comment]
Current dependencies on/for this PR:
* main
  * **PR #384** <a href="https://app.graphite.dev/github/pr/ourzora/zora-protocol/384?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #385** <a href="https://app.graphite.dev/github/pr/ourzora/zora-protocol/385?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #386** <a href="https://app.graphite.dev/github/pr/ourzora/zora-protocol/386?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #387** <a href="https://app.graphite.dev/github/pr/ourzora/zora-protocol/387?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/ourzora/zora-protocol/387?utm_source=stack-comment).

File: packages/1155-contracts/src/delegation/ZoraCreator1155Attribution.sol [hunk 1]
```diff
@@ -156,23 +162,17 @@ library ZoraCreator1155Attribution {
             "CreatorAttribution(TokenCreationConfig tokenConfig,uint32 uid,uint32 version,bool deleted)TokenCreationConfig(string tokenURI,uint256 maxSupply,uint64 maxTokensPerAddress,uint96 pricePerToken,uint64 mintStart,uint64 mintDuration,uint32 royaltyMintSchedule,uint32 royaltyBPS,address royaltyRecipient,address fixedPriceMinter)"
         );
 
-    function hashPremint(PremintConfig memory premintConfig) internal pure returns (bytes32) {
-        return
-            keccak256(
-                abi.encode(ATTRIBUTION_DOMAIN_V1, _hashToken(premintConfig.tokenConfig), premintConfig.uid, premintConfig.version, premintConfig.deleted)
-            );
+    function hashPremint(TokenCreationConfig memory tokenConfig, PremintSettings memory settings) internal pure returns (bytes32) {
```
Comment (oveddan) [inline packages/1155-contracts/src/delegation/ZoraCreator1155Attribution.sol:165]: this file change doens't belong in this PR

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -2,7 +2,7 @@ name: Lint
 on: [pull_request]
 
 jobs:
-  inspect-storage:
+  lint:
```
Comment (iainnash) [inline .github/workflows/lint.yml:5]: should this be another PR?

Comment (oveddan) [reply .github/workflows/lint.yml:5]: yes - will do

Comment (oveddan) [reply .github/workflows/lint.yml]: done in https://github.com/ourzora/zora-protocol/pull/391

File: packages/protocol-sdk/src/premint/preminter.ts [hunk 3]
```diff
@@ -1,72 +1,409 @@
...
+  typeof preminterAbi,
+  "premintV2"
+>["inputs"];
+
+type PremintV2HashDataTypes = AbiParametersToPrimitiveTypes<PremintV2Inputs>;
+
+export type PremintConfigV2 = PremintV2HashDataTypes[1];
+export type TokenCreationConfigV2 = PremintConfigV2["tokenConfig"];
+
+const v1Types = {
```
Comment (iainnash) [inline packages/protocol-sdk/src/premint/preminter.ts:46]: nit: Should we put this into another file?
